src/auth_v3.js

```diff
@@ -281,8 +281,8 @@ export function oauthFlow (cozy, storage, clientParams, onRegistered) {
           //
           // If it is an explicit Unauthorized error though, we bail, clear th
           // cache and retry.
-          if (FetchError.isUnauthorized(err)) {
-            throw err
+          if (FetchError.isUnauthorized(err) || FetchError.isNotFound(err)) {
+            throw new Error('Client has been revoked')
```

Comment:
I don't consider that a revoked client is a `FetchError`: it is not really tied to a single request to the stack, but something more global.